File: src/Entity/Randonnee.php
<?php

namespace App\Entity;
use App\Entity\Useer;
use DateTime;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Symfony\Component\Form\Extension\Core\Type\DateType;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Validator\Constraints\Image;
use App\Repository\RandonneeRepository;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity(repositoryClass=RandonneeRepository::class)
 */
class Randonnee
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\Length(
     * min = 3,
     * max = 255,
     * minMessage = "Erreur il doit comporter au moins {{ limit }} caractères",
     * maxMessage = "Erreur il  doit comporter au plus {{ limit }} caractères"
     * )
     */
    private $villedepart;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\Length(
     * min = 3,
     * max = 255,
     * minMessage = "Erreur il doit comporter au moins {{ limit }} caractères",
     * maxMessage = "Erreur il  doit comporter au plus {{ limit }} caractères"
     * )
     */
    private $villearrivee;

    /**
     * @ORM\Column(type="datetime",nullable=true)
     */
    private $datedepart;

    /**
     * @ORM\Column(type="datetime",nullable=true)
     */
    private $dateretour;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $activite;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $description;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $image;

    /**
     * @ORM\Column(type="integer")
     */
    private $duree;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $difficulte;

    /**
     * @ORM\Column(type="integer")
     */
    private $budget;

    /**
     * @ORM\OneToMany(targetEntity=Organisation::class, mappedBy="activite")
     */
    private $organisations;

    public function __construct()
    {
        $this->organisations = new ArrayCollection();
    }

    /**
     * @ORM\OneToMany(targetEntity=PostLiike::class, mappedBy="post")

    private $likes;

    public function __construct()
    {
        $this->likes = new ArrayCollection();
    }*/

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getVilledepart(): ?string
    {
        return $this->villedepart;
    }

    public function setVilledepart(string $villedepart): self
    {
        $this->villedepart = $villedepart;

        return $this;
    }

    public function getVillearrivee(): ?string
    {
        return $this->villearrivee;
    }

    public function setVillearrivee(string $villearrivee): self
    {
        $this->villearrivee = $villearrivee;

        return $this;
    }

    /**
     * @return mixed
     */
    public function getDatedepart()
    {
        return $this->datedepart;
    }

    /**
     * @param mixed $datedepart
     */
    public function setDatedepart($datedepart): void
    {
        $this->datedepart = $datedepart;
    }

    /**
     * @return mixed
     */
    public function getDateretour()
    {
        return $this->dateretour;
    }

    /**
     * @param mixed $dateretour
     */
    public function setDateretour($dateretour): void
    {
        $this->dateretour = $dateretour;
    }


    public function getActivite(): ?string
    {
        return $this->activite;
    }

    public function setActivite(string $activite): self
    {
        $this->activite = $activite;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getImage()
    {
        return $this->image;
    }

    public function setImage( $image)
    {
        $this->image = $image;

        return $this;
    }

    public function getDuree(): ?int
    {
        return $this->duree;
    }

    public function setDuree(int $duree): self
    {
        $this->duree = $duree;

        return $this;
    }

    public function getDifficulte(): ?string
    {
        return $this->difficulte;
    }

    public function setDifficulte(string $difficulte): self
    {
        $this->difficulte = $difficulte;

        return $this;
    }

    public function getBudget(): ?int
    {
        return $this->budget;
    }

    public function setBudget(int $budget): self
    {
        $this->budget = $budget;

        return $this;
    }

    /**
     * @return Collection|PostLiike[]

    public function getLikes(): Collection
    {
        return $this->likes;
    }

    public function addLike(PostLiike $like): self
    {
        if (!$this->likes->contains($like)) {
            $this->likes[] = $like;
            $like->setPost($this);
        }

        return $this;
    }

    public function removeLike(PostLiike $like): self
    {
        if ($this->likes->removeElement($like)) {
            // set the owning side to null (unless already changed)
            if ($like->getPost() === $this) {
                $like->setPost(null);
            }
        }

        return $this;
    }*/

    /**
     * permet de savoir si cette randonnée est liké par un utilisateur
     * @param Useer $useer
     * @return bool

    public function isLikedByUser(Useer $useer) : bool {
    foreach ($this->likes as $like){
        if($like->getUseer() == $useer) return true;
    }
    return false;

    } */

    /**
     * @return Collection|Organisation[]
     */
    public function getOrganisations(): Collection
    {
        return $this->organisations;
    }

    public function addOrganisation(Organisation $organisation): self
    {
        if (!$this->organisations->contains($organisation)) {
            $this->organisations[] = $organisation;
            $organisation->setActivite($this);
        }

        return $this;
    }

    public function removeOrganisation(Organisation $organisation): self
    {
        if ($this->organisations->removeElement($organisation)) {
            // set the owning side to null (unless already changed)
            if ($organisation->getActivite() === $this) {
                $organisation->setActivite(null);
            }
        }

        return $this;
    }
}

File: src/Tfarhida/entities/randonne/Randonnee.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Tfarhida.entities.randonne;

import java.util.Date;
import java.util.Objects;
import javafx.beans.property.SimpleStringProperty;
import javafx.beans.property.SimpleIntegerProperty;

/**
 *
 * @author PC
 */
public class Randonnee {
 private  SimpleIntegerProperty id;   
 private  SimpleStringProperty villedepart;
 private  SimpleStringProperty villearrivee;
 private  SimpleStringProperty datedepart;
 private  SimpleStringProperty dateretour;
 private  SimpleStringProperty activite;
 private  SimpleStringProperty description;
 private  SimpleStringProperty image;
 private  SimpleIntegerProperty duree;
 private  SimpleStringProperty difficulte;
 private  SimpleIntegerProperty budget;

    public Randonnee() {
    }

    public Randonnee(int id,String villedepart, String villearrivee, String datedepart, String daterretour, String activite, String description, String image, int duree, String difficulte, int budget) {
        this.id = new SimpleIntegerProperty(id);
        this.villedepart = new SimpleStringProperty(villedepart);
        this.villearrivee = new SimpleStringProperty(villearrivee);
        this.datedepart = new SimpleStringProperty(datedepart);
        this.dateretour = new SimpleStringProperty(daterretour);
        this.activite = new SimpleStringProperty(activite);
        this.description = new SimpleStringProperty(description);
        this.image = new SimpleStringProperty(image);
        this.duree = new SimpleIntegerProperty(duree);
        this.difficulte = new SimpleStringProperty(difficulte);
        this.budget = new SimpleIntegerProperty(budget);
    }
    public Randonnee(String villedepart, String villearrivee, String datedepart, String daterretour, String activite, String description, String image, int duree, String difficulte, int budget) {
        this.villedepart = new SimpleStringProperty(villedepart);
        this.villearrivee = new SimpleStringProperty(villearrivee);
        this.datedepart = new SimpleStringProperty(datedepart);
        this.dateretour = new SimpleStringProperty(daterretour);
        this.activite = new SimpleStringProperty(activite);
        this.description = new SimpleStringProperty(description);
        this.image = new SimpleStringProperty(image);
        this.duree = new SimpleIntegerProperty(duree);
        this.difficulte = new SimpleStringProperty(difficulte);
        this.budget = new SimpleIntegerProperty(budget);
    }
    
     public int getId() {
        return id.get();
    }

    public void setId(int id) {
        this.id = new SimpleIntegerProperty(id);
    }

    public String getVilledepart() {
        return villedepart.get();
    }

    public void setVilledepart(String villedepart) {
        this.villedepart = new SimpleStringProperty(villedepart);
    }

    public String getVillearrivee() {
        return villearrivee.get();
    }

    public void setVillearrivee(String villearrivee) {
        this.villearrivee = new SimpleStringProperty(villearrivee);
    }

    public String getDatedepart() {
        return datedepart.get();
    }

    public void setDatedepart(String datedepart) {
        this.datedepart = new SimpleStringProperty(datedepart);
    }

    public String getDateretour() {
        return dateretour.get();
    }

    public void setDateretour(String daterretour) {
        this.dateretour = new SimpleStringProperty(daterretour);
    }

    public String getActivite() {
        return activite.get();
    }

    public void setActivite(String activite) {
        this.activite = new SimpleStringProperty(activite);
    }

    public String getDescription() {
        return description.get();
    }

    public void setDescription(String description) {
        this.description = new SimpleStringProperty(description);
    }

    public String getImage() {
        return image.get();
    }

    public void setImage(String image) {
        this.image = new SimpleStringProperty(image);
    }

    public int getDuree() {
        return duree.get();
    }

    public void setDuree(int duree) {
        this.duree = new SimpleIntegerProperty(duree);
    }

    public String getDifficulte() {
        return difficulte.get();
    }

    public void setDifficulte(String difficulte) {
        this.difficulte = new SimpleStringProperty(difficulte);
    }

    public int getBudget() {
        return budget.get();
    }

    public void setBudget(int budget) {
        this.budget = new SimpleIntegerProperty(budget);
    }

    public int hashCode() {
        int hash = 5;
        hash = 97 * hash + Objects.hashCode(this.villedepart);
        hash = 97 * hash + Objects.hashCode(this.villearrivee);
        hash = 97 * hash + Objects.hashCode(this.datedepart);
        hash = 97 * hash + Objects.hashCode(this.dateretour);
        hash = 97 * hash + Objects.hashCode(this.activite);
        hash = 97 * hash + Objects.hashCode(this.description);
        hash = 97 * hash + Objects.hashCode(this.image);
        hash = 97 * hash + Objects.hashCode(this.difficulte);
        return hash;
    }

    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Randonnee other = (Randonnee) obj;
        if (this.id != other.id) {
            return false;
        }
        if (this.duree != other.duree) {
            return false;
        }
        if (this.budget != other.budget) {
            return false;
        }
        if (!Objects.equals(this.villedepart, other.villedepart)) {
            return false;
        }
        if (!Objects.equals(this.villearrivee, other.villearrivee)) {
            return false;
        }
        if (!Objects.equals(this.activite, other.activite)) {
            return false;
        }
        if (!Objects.equals(this.description, other.description)) {
            return false;
        }
        if (!Objects.equals(this.image, other.image)) {
            return false;
        }
        if (!Objects.equals(this.difficulte, other.difficulte)) {
            return false;
        }
        if (!Objects.equals(this.datedepart, other.datedepart)) {
            return false;
        }
        if (!Objects.equals(this.dateretour, other.dateretour)) {
            return false;
        }
        return true;
    }

    public String toString() {
        return "Randonnee{" + "id=" + id + ", villedepart=" + villedepart + ", villearrivee=" + villearrivee + ", datedepart=" + datedepart + ", daterretour=" + dateretour + ", activite=" + activite + ", description=" + description + ", image=" + image + ", duree=" + duree + ", difficulte=" + difficulte + ", budget=" + budget + '}';
    }

   
    
}
